
Move to regparm 1 for nested functions.  Fixes CPU context corruption
in fs/xfs.c and possibly other places.

--- aclocal.m4~	2007-11-25 03:01:30.000000000 +0100
+++ aclocal.m4	2008-02-01 16:47:36.000000000 +0100
@@ -333,13 +333,13 @@
 
 if test "x$grub_cv_i386_check_nested_functions" = xyes; then
   AC_DEFINE([NESTED_FUNC_ATTR], 
-	[__attribute__ ((__regparm__ (2)))],
+	[__attribute__ ((__regparm__ (1)))],
 	[Catch gcc bug])
 else
 dnl Unfortunately, the above test does not detect a bug in gcc-4.0.
 dnl So use regparm 2 until a better test is found.
   AC_DEFINE([NESTED_FUNC_ATTR], 
-	[__attribute__ ((__regparm__ (2)))],
+	[__attribute__ ((__regparm__ (1)))],
 	[Catch gcc bug])
 fi
 ])
